shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs

```diff
@@ -67,7 +68,7 @@ ledgersTransition = do
 
   let UTxOState utxo' dep fee (ppup, aup, favs, avs) = u''
   let (favs', ready) = Map.partitionWithKey (\s _ -> s > slot) favs
-  let avs' = newAVs avs ready
+  let avs' = Applications $ apps avs ⨃ (Map.toList . apps $ newAVs avs ready)
```

Comment:
unless this is a rare manipulation of `Applications`, it might be worth making `Applications` an instance of `Relation`, then the above could be more like:

`avs ⨃ (newAVs avs ready)`